server/sonar-server/src/main/java/org/sonar/server/plugins/ws/PluginWSCommons.java

```diff
@@ -81,11 +87,17 @@
   public static final Comparator<PluginUpdate> NAME_KEY_PLUGIN_UPDATE_ORDERING = Ordering.from(NAME_KEY_PLUGIN_ORDERING)
     .onResultOf(PluginUpdateToPlugin.INSTANCE);
 
-  void writePluginInfo(JsonWriter json, PluginInfo pluginInfo, @Nullable String category) {
+  void writePluginInfo(JsonWriter json, PluginInfo pluginInfo, @Nullable String category, @Nullable PluginDto pluginDto) {
     json.beginObject();
 
     json.prop(PROPERTY_KEY, pluginInfo.getKey());
     json.prop(PROPERTY_NAME, pluginInfo.getName());
+    if (pluginDto != null) {
```

Comment:
In which case `pluginDto` is null ? That would mean that the plugin has not been registered in db. I suggest to fail-fast in the caller of this method.

Reply:
it will be null for api/plugins/pending. It will fail fast if it's not found for api/plugins/installed.